nvdaHelper/UIARemote/UIARemote.cpp

```diff
@@ -57,26 +56,38 @@ extern "C" __declspec(dllexport) bool __stdcall msWord_getCustomAttributeValue(I
 		UiaTextRange textRange{pTextRangeArg};
 		UiaInt customAttribID{customAttribIDArg};
 		UiaVariant customAttribValue;
-		scope.If(docElement.IsExtensionSupported(guid_msWord_extendedTextRangePattern),[&]() {
-			logger<<L"guid_msWord_extendedTextRangePattern is supported extension"<<endl;
-			UiaElement patternElement{nullptr};
-			docElement.CallExtension(guid_msWord_extendedTextRangePattern, patternElement);
-			scope.If(patternElement,[&]() {
-				logger<<L"Got custom pattern element "<<endl;
-				scope.If(patternElement.IsExtensionSupported(guid_msWord_getCustomAttributeValue),[&]() {
-					isExtensionSupported = true;
-					logger<<L"guid_msWord_getCustomAttributeValue extension supported on pattern"<<endl;
-					patternElement.CallExtension(guid_msWord_getCustomAttributeValue, textRange, customAttribID, customAttribValue);
-					logger<<L"Called guid_msWord_getCustomAttributeValue extention"<<endl;
-				}, [&]() {
-					logger<<L"No guid_msWord_getCustomAttributeValue extension supported"<<endl;
-				});
-			}, [&]() {
-				logger<<L"Could not fetch guid_msWord_extendedTextRangePattern pattern"<<endl;
-			});
-		}, [&]() {
-			logger<<L"No guid_msWord_extendedTextRangePattern extension supported"<<endl;
-		});
+		scope.If(
+			/* condition */ docElement.IsExtensionSupported(guid_msWord_extendedTextRangePattern),
+			/* body */ [&]() {
+				logger<<L"guid_msWord_extendedTextRangePattern is supported extension"<<endl;
+				UiaElement patternElement{nullptr};
+				docElement.CallExtension(guid_msWord_extendedTextRangePattern, patternElement);
+				scope.If(
+					/* condition */ patternElement,
+					/* body */ [&]() {
+						logger<<L"Got custom pattern element "<<endl;
+						scope.If(
+							/* condition */ patternElement.IsExtensionSupported(guid_msWord_getCustomAttributeValue),
+							/* body */ [&]() {
+								isExtensionSupported = true;
+								logger<<L"guid_msWord_getCustomAttributeValue extension supported on pattern"<<endl;
+								patternElement.CallExtension(guid_msWord_getCustomAttributeValue, textRange, customAttribID, customAttribValue);
+								logger<<L"Called guid_msWord_getCustomAttributeValue extention"<<endl;
+							},
+							/* else */ [&]() {
+								logger<<L"No guid_msWord_getCustomAttributeValue extension supported"<<endl;
+							}
+						);
+					},
+					/* else */ [&]() {
+						logger<<L"Could not fetch guid_msWord_extendedTextRangePattern pattern"<<endl;
+					}
+				);
+			},

```

Comment:
I don't see how that can be true (yet!). It makes sense that this would happen in the `Resolve` call, but I think that in this case it can only result in two cross platform calls. It won't know about the second `scope.if` until the lambda is executed, which must happen locally. It can't execute the lambda until it has the result of the first condition.

I suspect that within `Resolve` when an additional `scope.if` is called from a handler, it is recursively resolved immediately.

Reply:
After chatting about this. The missing piece is that both the body and the else lambda's are executed immediately with the goal of constructing remote ops instructions. This means the instruction tree produced can be run in its entirety on in the remote process, ensuring that all logic is executed with a single process call. The log message are recorded on both sides of the condition, but only emitted based on the result of the condition.

Note: this means that modifying any local (non-remote ops type) variable within these lambdas is likely a bug. 
We'll have to watch out for local variables sneaking into these lambdas.